src/relax/op/tensor/manipulate.cc

```diff
@@ -88,21 +88,22 @@ StructInfo InferStructInfoBroadcastTo(const Call& call, const BlockBuilder& ctx)
     const auto* old_len_int = old_len.as<IntImmNode>();
     if (old_len_int != nullptr && old_len_int->value == 1) {
       continue;
-    } else if (!analyzer->CanProveEqual(old_len, tgt_len)) {
```

Comment:
Ah yeah this portion of code is specifically requiring shapes to be known and checking if those static shapes are compatible for broadcasting. My opinion is that we should not enforce that since dynamic broadcast_to shows up pretty frequently. This should be fully removed rather than commented out though.

Reply:
I see. The intent of the original code was to require using a `MatchCast` to check the shape first. However, I agree with you that the operator implementation can do the checking, so that's unnecessary work for the user